extension/packages/nextjs/app/greetings/page.tsx

```diff
@@ -0,0 +1,65 @@
+"use client";
+
+import type { NextPage } from "next";
+import { gql, useQuery } from "urql";
```

Comment:
Maybe we can use [Tanstack query](https://tanstack.com/query/v5/docs/framework/react/graphql) since we already have it? Or `urql` is recommended?

Reply:
Yes! Sure, makes sense! I don't know why I used urql...

Reply:
@rin-st I don't know too much about TanStack Query, do you know if there is an easy way to parse the GraphQL query?

I mean, to do what I'm doing here https://github.com/scaffold-eth/create-eth-extensions/pull/12/files#diff-4629708b76d08bf9ab81b2b163d0f730144ad954df6d16b91b55ceb0a140c89dR9 with gql.

The example from TanStack use a user-defined function https://github.com/dotansimha/graphql-code-generator/blob/7c25c4eeb77f88677fd79da557b7b5326e3f3950/examples/front-end/react/tanstack-react-query/src/gql/gql.ts#L14

Reply:
I changed `urql` to `react-query`, check it here [fcf33e3](https://github.com/scaffold-eth/create-eth-extensions/pull/12/commits/fcf33e3185e77748cbed6e75b8698a8b2c6fb95f) . 

I tried also to import `Greeting` type from `ponder`'s generated `Schema` type (I reexported what is defined in ponder-dev.d.ts), but for some reason it doesn't work for me. Ts understand it is as `any`. Probably it's because it also doesn't understand ponder's tsconfig for me. So types are just copypasted for now to `greetings` page